subsys/bluetooth/host/gatt.c

```diff
@@ -1019,11 +1021,16 @@ static u8_t gatt_foreach_iter(const struct bt_gatt_attr *attr,
 		return BT_GATT_ITER_CONTINUE;
 	}
 
-	if (*num_matches) {
+	if (num_matches) {
 		*num_matches -= 1;
 	}
 
-	return func(attr, user_data);
+	result = func(attr, user_data);
+	if (result == BT_GATT_ITER_STOP || (num_matches && !*num_matches)) {
```

Comment:
If result is ITER_STOP, we will return it anyway.
Simplify it to this?
```
if (num_matches && !*num_matches) {
  return BT_GATT_ITER_STOP;
}
return result;
```

Reply:
One more instance of checking for num_matches for NULL before deref.
This could be:
```
if (result == BT_GATT_ITER_STOP || !*num_matches)
```
or even:
```
if (!*num_matches){
  return BT_GATT_ITER_STOP;
}
return result
```